dagshub/data_engine/model/query_result.py

```diff
@@ -418,6 +419,8 @@ def _get_blob_fn(dp: Datapoint, field: str, url: str, blob_path: Path):
 
         # Convert any downloaded document fields
         document_fields = [f for f in fields if f in self.document_fields]
+        # Exclude any annotation fields, because they are already converted above
+        document_fields = [f for f in document_fields if f not in self.annotation_fields]
```

Comment:
It would make more sense to me for every processed field to either be loaded as document or annotation here in this function, and not called twice from outside and then filter out annotations when it's called for documents.

In the current flow if we were to add a new tag that conflicts with the existing one, you would need to double exclude every field from the ones they don't belong to.

Let me know if this fix is easy to make.